q2_diversity/tests/test_beta.py

```diff
@@ -114,24 +115,31 @@ def test_beta_empty_table(self):
     def test_beta_phylogenetic(self):
         t = self.two_feature_table
         tree = self.three_feature_tree
-        actual = self.beta_phylogenetic(
-            table=t, phylogeny=tree, metric='unweighted_unifrac')
 
-        self.assertEqual(len(actual), 1)
+        for method, table_, tree_ in ((self.beta_phylogenetic, t, tree),
+                                      (self.beta_phylogenetic_meta, [t, t],
+                                       [tree, tree])):
+            with self.subTest(method=method):
+                actual = method(
+                    table=table_, phylogeny=tree_, metric='unweighted_unifrac')
 
-        self.assertEqual(repr(actual.distance_matrix.type), 'DistanceMatrix')
+                self.assertEqual(len(actual), 1)
 
-        # expected computed with skbio.diversity.beta_diversity
-        expected = skbio.DistanceMatrix([[0.00, 0.25, 0.25],
-                                         [0.25, 0.00, 0.00],
-                                         [0.25, 0.00, 0.00]],
-                                        ids=['S1', 'S2', 'S3'])
+                self.assertEqual(repr(actual.distance_matrix.type),
+                                 'DistanceMatrix')
 
-        actual = actual[0].view(skbio.DistanceMatrix)
-        self.assertEqual(actual.ids, expected.ids)
-        for id1 in actual.ids:
-            for id2 in actual.ids:
-                npt.assert_almost_equal(actual[id1, id2], expected[id1, id2])
+                # expected computed with skbio.diversity.beta_diversity
+                expected = skbio.DistanceMatrix([[0.00, 0.25, 0.25],
+                                                 [0.25, 0.00, 0.00],
+                                                 [0.25, 0.00, 0.00]],
+                                                ids=['S1', 'S2', 'S3'])
+
+                actual = actual[0].view(skbio.DistanceMatrix)
+                self.assertEqual(actual.ids, expected.ids)
+                for id1 in actual.ids:
+                    for id2 in actual.ids:
+                        npt.assert_almost_equal(actual[id1, id2],
+                                                expected[id1, id2])
```

Comment:
Could we refactor this into a test helper?

Reply:
Sure. Note that most of this is not new, just indented, but it could definitely benefit from some refactor. Will clean up